xcm/xcm-executor/src/lib.rs

```diff
@@ -401,13 +400,21 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				for asset in deposited.assets_iter() {
 					Config::AssetTransactor::deposit_asset(&asset, &dest)?;
 				}
-				let assets = Self::reanchored(deposited, &dest)?;
+				// Note that we pass `None` as `maybe_failed_bin` and drop any assets which cannot
+				// be reanchored  because we have already called `deposit_asset` on all assets.
```

Comment:
Security-wise, this is fine, because the funds are already in the sovereign account of the destination. It's just a bit unfortunate that the destination will have no notification of the assets that were received in its sovereign account on the reserve, and I don't really have a solution for that.

Reply:
I considered it and I don't think there is an easy one which doesn't involve introducing substantial code complexity and performance loss.